code/game/machinery/doors/firedoor.dm

```diff
@@ -532,6 +601,10 @@
 			new /obj/item/electronics/firelock (targetloc)
 	qdel(src)
 
+/obj/machinery/door/firedoor/Moved(atom/oldloc)
+	. = ..()
+	unregister_adjacent_turfs(oldloc)
+
```

Comment:
do you want to register for new adjacent turfs here?

Reply:
ah shoot